samples/subsys/canbus/canopen/README.rst

```diff
@@ -0,0 +1,350 @@
+This sample application shows how the `CANopenNode`_ CANopen protocol
+stack can be used in Zephyr.
+
+CANopen is an internationally standardized (`EN 50325-4`_, `CiA 301`_)
+communication protocol and device specification for embedded
+systems used in automation. CANopenNode is a 3rd party, open-source
+CANopen protocol stack.
+
+Apart from the CANopen protocol stack integration, this sample also
+demonstrates the use of non-volatile storage for the CANopen object
+dictionary.
+
+Requirements
+************
+
+* A board with CAN bus and flash support
+* Host PC with CAN bus support
+
+Building and Running
+********************
+
+Building and Running for TWR-KE18F
+==================================
+The :ref:`twr_ke18f` board is equipped with an onboard CAN
+transceiver. This board supports CANopen LED indicators (red and green
+LEDs). The sample can be built and executed for the TWR-KE18F as
+follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/canbus/canopen
+   :board: twr_ke18f
+   :goals: build flash
+   :compact:
+
+Pressing the button labelled ``SW3`` will increment the button press
+counter object at index ``0x2102`` in the object dictionary.
+
+Building and Running for FRDM-K64F
+==================================
+The :ref:`frdm_k64f` board does not come with an onboard CAN
+transceiver. In order to use the CAN bus on the FRDM-K64F board, an
+external CAN bus tranceiver must be connected to ``PTB18``
+(``CAN0_TX``) and ``PTB19`` (``CAN0_RX``). This board supports CANopen
+LED indicators (red and green LEDs)
+
+The sample can be built and executed for the FRDM-K64F as follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/canbus/canopen
+   :board: frdm_k64f
+   :goals: build flash
+   :compact:
+
+Pressing the button labelled ``SW3`` will increment the button press
+counter object at index ``0x2102`` in the object dictionary.
+
+Testing CANopen Communication
+*****************************
+CANopen communication between the host PC and Zephyr can be
+established using any CANopen compliant application on the host PC.
+The examples here uses `CANopen for Python`_ for communicating between
+the host PC and Zephyr.  First, install python-canopen along with the
+python-can backend as follows:
+
+.. code-block:: console
+
+   pip3 install --user canopen can
+
+Next, bring up the CAN interface on the test PC. On GNU/Linux, this
+can be done as follows:
+
+.. code-block:: console
+
+   sudo ip link set can0 type can bitrate 125000 restart-ms 100
+   sudo ip link set up can0
+
+To better understand the communication taking place in the examples
+below, it might be beneficial to monitor the CAN traffic from the host
+PC. On GNU/Linux, this can be accomplished using ``candump`` from the
+`can-utils`_ package as follows:
+
+.. code-block:: console
+
+   candump can0
+
+NMT State Changes
+=================
+Changing the Network Management (NMT) state of the node can be
+accomplished using the following python code:
```

Comment:
Python (the language) is a proper name and should be capitalized.  Check for more uses of "python" that should be "Python".

Reply:
@dbkinder Thanks, fixed all over.